pypesto/C.py

```diff
@@ -0,0 +1,127 @@
+"""
+Constants
+=========
+Package-wide consistent constant definitions.
+"""
+
+from enum import Enum
+
+from typing import Tuple, Union
+
+###############################################################################
+# OBJECTIVE
+
+MODE_FUN = 'mode_fun'  # mode for function values
+MODE_RES = 'mode_res'  # mode for residuals
+FVAL = 'fval'  # function value
+FVAL0 = 'fval0'  # function value at start
+GRAD = 'grad'  # gradient
+HESS = 'hess'  # Hessian
+HESSP = 'hessp'  # Hessian vector product
+RES = 'res'  # residual
+SRES = 'sres'  # residual sensitivities
+RDATAS = 'rdatas'  # returned simulated data sets
+
+TIME = 'time'  # time
+N_FVAL = 'n_fval'  # number of function evaluations
+N_GRAD = 'n_grad'  # number of gradient evaluations
+N_HESS = 'n_hess'  # number of Hessian evaluations
+N_RES = 'n_res'  # number of residual evaluations
+N_SRES = 'n_sres'  # number of residual sensitivity evaluations
+CHI2 = 'chi2'  # chi2 value
+SCHI2 = 'schi2'  # chi2 value gradient
+X = 'x'
```

Comment:
Might conflict with usage of `X` as a vector of `x` in some scripts

Reply:
If done like PEtab constants, then this might be renamed more informatively as e.g. `PARAMETER_VECTOR` (but currently consistent with the other shortened constant names, so fine as is / out of scope...)

Reply:
definitely a good point, we should consider